code/game/objects/items/weapons/swords_axes_etc.dm

```diff
@@ -97,10 +97,26 @@
 		if(..() == 1)
 			playsound(src.loc, "swing_hit", 50, 1, -1)
 			if (target_zone == "r_leg" || target_zone == "l_leg")
+				var/stun_chance = 100
 				if(ishuman(target))
 					var/mob/living/carbon/human/T = target
 					var/armor = T.run_armor_check(T, "melee")
 					if(armor < 100)
+						stun_chance -= 30
+
+					if(T.shoes && T.shoes.item_flags & NOSLIP)
+						stun_chance -= 25
+
+					if(T.species.flags & NO_SLIP)
+						stun_chance -= 25
+
+					if(T.species.brute_mod<0.8)
+						stun_chance -= 25
+
+					if(T.wear_suit && istype(T.wear_suit,/obj/item/clothing/suit/space))
+						stun_chance -= 30
+
+					if(prob(stun_chance))

```

Comment:
The stun chance above can become negative.
The DM reference doesn't say anything about the valid range of the argument, so I would change it to that to make sure there are not unwanted side effects:
`prob(max(0,stun_chance))`

Reply:
I found out that negative chances is equal to 0, neither it seems to cause a runtime.